src/grid/cubic.py

```diff
@@ -0,0 +1,582 @@
+                        )
+                        for i in range(1, nu_z + 1)
+                    ]
+                    deriv_var = nu_z
+                # Sympy symbols and dictionary of symbols pointing to the derivative values
+                sympy_symbols = symbols("x:" + str(deriv_var))
+                symbol_values = {
+                    "x" + str(i): float(derivs[i]) for i in range(0, deriv_var)
+                }
+                return interpolated * float(
+                    sum(
+                        [
+                            bell(deriv_var, i, sympy_symbols).evalf(subs=symbol_values)
+                            for i in range(1, deriv_var + 1)
+                        ]
+                    )
+                )
+            else:
+                raise NotImplementedError(
+                    "Taking mixed derivative while applying the logarithm is not supported."
+                )
+        # Normal interpolation without logarithm.
+        interpolated = x_spline(real_pt[0], real_pt[1], real_pt[2], nu_x)
+        return interpolated
+
+    def coordinates_to_index(self, indices):
+        r"""Convert (i, j, k) integer coordinates to the grid index m.
+
+        Parameters
+        ----------
+        indices : (int, int, int)
+            The ith, jth, kth position of the grid point.
+
+        Returns
+        -------
+        index : int
+            Index of the grid point.
+
+        """
+        n_1d, n_2d = self.shape[2], self.shape[1] * self.shape[2]
+        index = n_2d * indices[0] + n_1d * indices[1] + indices[2]
+        return index
+
+    def index_to_coordinates(self, index):
+        r"""Convert index of grid point to its (i, j, k) coordinates in a cubic grid.
+
+        Cubic grid has a shape of :math:`(N_x, N_y, N_z)` denoting the number of points in
+        :math:`x`, :math:`y`, and :math:`z` directions. So, each grid point has a :math:`(i, j, k)`
+        integer coordinate where :math:`0 <= i <= N_x - 1`, :math:`0 <= j <= N_y - 1`,
+        and :math:`0 <= k <= N_z - 1`.
+
+        Parameters
+        ----------
+        index : int
+            Index of the grid point.
+
+        Returns
+        -------
+        indices : (int, int, int)
+            The corresponding :math:`(i, j, k)` integer coordinates in a cubic grid.
+
+        """
+        if not index >= 0:
+            raise ValueError(
+                f"Argument index should be a positive integer, got {index}"
+            )
+        n_1d, n_2d = self.shape[2], self.shape[1] * self.shape[2]
+        i = index // n_2d
+        j = (index - n_2d * i) // n_1d
+        k = index - n_2d * i - n_1d * j
+        return i, j, k
+
+
+class Tensor1DGrids(_CubicGrid):
+    r"""Tensor1DGrids : Tensor product of three one-dimensional grids."""
+
+    def __init__(self, oned_x, oned_y, oned_z):
+        r"""Construct Tensor1DGrids by tensor product of three one-dimensional grids.
+
+        Parameters
+        ----------
```

Comment:
@Ali-Tehrani Can you please complete the docstring? Also, I was thinking that `oned_z` can be optional (with a default value of `None`) in which case a 2D grid is made. Can you please add that?